app/qml/map/MMMapController.qml

```diff
@@ -92,50 +92,58 @@ Item {
   ]
 
   onStateChanged: {
-    // We call this first because when previous state is 'view' and `centeredToGPS` is true
-    // the map may still not be centered. It's only centered after a certain threshold is exceeded.
-    updatePosition()
-
     switch ( state ) {
+      case "record": {
+        if ( __appSettings.autolockPosition ) { // center to GPS
+          if ( gpsStateGroup.state === "unavailable" ) {
+              __notificationModel.addError( "GPS currently unavailable." )
+              return;
+          }
 
-    case "record": {
-      root.recordingStarted()
-      break
-    }
+          root.centeredToGPS = true
+        }
 
-    case "recordInLayer": {
-      root.recordInLayerFeatureStarted()
-      root.hideHighlight()
-      break
-    }
+        root.recordingStarted()
+        break
+      }
 
-    case "edit": {
-      root.editingGeometryStarted()
-      root.hideHighlight()
-      break
-    }
+      case "recordInLayer": {
+        root.recordInLayerFeatureStarted()
+        root.hideHighlight()
+        break
+      }
 
-    case "split": {
-      root.showInfoTextMessage( qsTr( "Create line to split the selected feature" ) )
-      root.splittingStarted()
-      break
-    }
+      case "edit": {
+        root.editingGeometryStarted()
+        root.hideHighlight()
+        break
+      }
 
-    case "view": {
-      root.hideHighlight()
-      break
-    }
+      case "split": {
+        root.showInfoTextMessage( qsTr( "Create line to split the selected feature" ) )
+        root.splittingStarted()
+        break
+      }
 
-    case "stakeout": {
-      root.hideHighlight()
-      root.stakeoutStarted( internal.stakeoutTarget )
-      break
-    }
+      case "view": {
+        root.hideHighlight()
+        break
+      }
 
-    case "inactive": {
-      break
-    }
+      case "stakeout": {
+        root.hideHighlight()
+        root.stakeoutStarted( internal.stakeoutTarget )
+        break
+      }
+
+      case "inactive": {
+        break
+      }
     }
+
+    // We call this first because when previous state is 'view' and `centeredToGPS` is true
+    // the map may still not be centered. It's only centered after a certain threshold is exceeded.
+    updatePosition()
```

Comment:
This comment is no longer true: we don't call this first, we actually call this last!
Does that mean we'll get a first point when `root.recordingStarted()` is called which may not be on the actual GPS position?
Why not call `updatePosition()` right after `root.centeredToGPS = true` ?